docs/rfcs/0005-coupons.md

```diff
@@ -0,0 +1,102 @@
+## Title for RFC
+
+coupons
+
+### Abstract
+
+This describes the various kinds of coupons and how they will be implemented.
+
+Note: For this RFC the "next" course run is the course run which the user would be enrolled
+in when they enroll in a course, whether it is currently running or will run soon.
+
+##### Discount
+
+Coupons may have a percent discount. There is no fixed discount use case at the moment.
+
+##### Basket
+
+There can only be one coupon per basket, and a user may add at most only one course run,
+course, or program to a basket. This simplifies the UX and means we don't need to consider
+how multiple coupons can conflict.
+
+Users ultimately purchase course runs. In the UI when a user buys a course it will add the
+next course run to the basket. When a user buys a program it will add the next course run 
+for each course to the basket.
+
+##### Coupon target
+
+Coupons may apply to programs, courses, or course runs. The logic for what targets a coupon
+is redeemable for is different from MicroMasters.
+
+A program coupon is redeemable only if all courses for the program are in the basket. If a course
+was missing from the basket, maybe because the program is nearing end of life and one of
+the courses is not enrollable anymore, then the program coupon would not be redeemable at all.
+  
+A course coupon would be redeemable only for the next course run for that course. It may have 
+multiple courses where it can be redeemed.
+
+A course run coupon is redeemable only for the particular course run.
+
+Note: The UI will not include a basket, the basket will only be in the backend. The end user
+will only be able to buy one course or one program at a time, and can apply only one coupon.
+This should simplify the UX for redeeming coupons.
+
+##### Number of redemptions
+
+Coupons have a limited number of total redemptions. They cannot be unlimited to ensure that
+there's a cap on the number of discounts applied.
+However this could be a very high number, higher than is likely to ever be redeemed.
+
+TODO: Does this sound right? If the number is actually unlimited we should handle it that way.
```

Comment:
We've confirmed that being able to set a limit is important. And there is general agreement that an arbitrarily large limit would be acceptable for coupons intended to be used by many people.

Reply:
:+1: